gpc-api-facade/src/main/java/uk/nhs/adaptors/pss/gpc/service/MigratePatientStructuredRecordService.java

```diff
@@ -0,0 +1,51 @@
+package uk.nhs.adaptors.pss.gpc.service;
+
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Identifier;
+import org.hl7.fhir.dstu3.model.Parameters;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+
+import lombok.AllArgsConstructor;
+import uk.nhs.adaptors.connector.dao.PatientMigrationRequestDao;
+import uk.nhs.adaptors.connector.model.RequestStatus;
+import uk.nhs.adaptors.pss.gpc.amqp.PssQueuePublisher;
+import uk.nhs.adaptors.pss.gpc.exception.FhirValidationException;
+
+@Component
+@AllArgsConstructor()
+public class MigratePatientStructuredRecordService {
+    private static final String PATIENT_NHS_NUMBER_PARAM_NAME = "patientNHSNumber";
+
+    private final FhirParseService fhirParseService;
+    private final PatientMigrationRequestDao patientMigrationRequestDao;
+    private final PssQueuePublisher pssQueuePublisher;
+
+    public ResponseEntity<String> handlePatientMigrationRequest(String body) {
```

Comment:
I think that business logic layer (service layer) should not be aware of HTTP responses. Top layer (controllers) should do that.